### PR TITLE
Update ci.yml

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
*Issue #, if available:*

*Description of changes:*
Build cedar-lean as part of CI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by khieta — APPROVED** [review]
Nice! Can you add `lake exe CedarUnitTests` too to run the tests?

**Review by aaronjeline — APPROVED** [review]

**Comment by andrewmwells-amazon** [comment]
> Nice! Can you add lake exe CedarUnitTests too to run the tests?

It seems like this process returns success even if a unit test fails. I'll try grepping for "FAILED" which catches the tests I've tried breaking. Let me know if that won't work in some cases.

**Comment by john-h-kastner-aws** [comment]
Feels like this should be easier to fix in Lean. Can `main` be `IO Nat`?

**Review by shaobo-he-aws — APPROVED** [review]

**Comment by andrewmwells-amazon** [comment]
Fails correctly on broken test: https://github.com/cedar-policy/cedar-spec/actions/runs/6723554552/job/18273878894?pr=144

File: cedar-lean/UnitTest/Decimal.lean [hunk 1]
```diff
@@ -29,7 +29,7 @@ private def testValid (str : String) (rep : Int) : TestCase :=
 def testsForValidStrings :=
   suite "Decimal.parse for valid strings"
   [
-    testValid "0.0" 0,
+    testValid "0.0." 0,
```
Comment (andrewmwells-amazon) [inline cedar-lean/UnitTest/Decimal.lean:32]:
```suggestion
    testValid "0.0" 0,
```

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**